src/main/java/com/o19s/es/ltr/ranker/normalizer/MinMaxFeatureNormalizer.java

```diff
@@ -0,0 +1,53 @@
+package com.o19s.es.ltr.ranker.normalizer;
+
+import org.elasticsearch.ElasticsearchException;
+
+/**
+ * MinMax Feature Normalization
+ * Generally following the standard laid out by sklearn:
+ *   (value / (max - min)) + min to give a normalized 0-1 feature value
+ *
+ * See
+ * https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MinMaxScaler.html
+ */
+public class MinMaxFeatureNormalizer implements Normalizer  {
+    float maximum;
+    float minimum;
+
+    public MinMaxFeatureNormalizer(float minimum, float maximum) {
+        if (minimum >= maximum) {
+            throw new ElasticsearchException("Minimum " + Double.toString(minimum) +
+                                             " must be smaller than than maximum: " +
+                                              Double.toString(maximum));
+        }
+        this.minimum = minimum;
+        this.maximum = maximum;
+    }
+
+    @Override
+    public float normalize(float value) {
+        return  (value - minimum) / (maximum - minimum);
```

Comment:
should this be guarded to avoid negative values?

Reply:
perhaps for min / max it should, as we should not have features with negative values logged from ES